docs/GDPs/gdp-0003-duck-storage.rst

```diff
@@ -0,0 +1,822 @@
+        + :code:`dtype: dtype_like`
+          The dtype of the storage (NumPy dtype or accepted by :code:`np.dtype()`). It defaults to
+          :code:`np.float64`.
+
+        + :code:`fill_value: Number`. The number to which the storage is initialized.
+
+        + :code:`shape: Sequence[int]`
+          Sequence of length :code:`ndim` (:code:`ndim` = number of dimensions, maximum 3) with the
+          shape of the storage, that is, the full addressable space in the allocated memory buffer.
+          (See also Section :ref:`domain_and_halo`)
+
+    For common keyword-only arguments, please see below.
+
+:code:`full_like(shape: Sequence[int], fill_value: Number, dtype=np.float64, **kwargs) -> Storage`
+    Allocate a storage with values initialized to the scalar given in :code:`fill_value`, while
+    choosing the not explicitly overridden parameters according to :code:`data`.
+
+    Parameters:
+        + :code:`data: Storage` Not explicitly overridden parameters are chosen as the value used in
+          this :code:`Storage`
+
+        + :code:`dtype: dtype_like`
+          The dtype of the storage (NumPy dtype or accepted by :code:`np.dtype()`). It defaults to
+          :code:`data.dtype`
+
+        + :code:`fill_value: Number`. The number to which the storage is initialized.
+
+    The common keyword-only arguments can also be overridden. Please see below for their description.
+
+    Note that :code:`shape` is not a parameter and can not be overridden, implying that also the
+    :code:`axes` can not be overridden.
+
+:code:`as_storage(data: array_like = None, device_data: array_like = None, *, sync_state: Storage.SyncState = None, **kwargs) -> Storage`
+    Wrap an existing buffer in a GT4Py storage instance, without copying the buffer's contents.
+
+    Parameters:
+        + :code:`data: array_like`. The memory buffer or storage from which the storage is
+          initialized.
+
+        + :code:`device_data: array_like`. The device buffer or storage in case wrapping
+          existing buffers on both the device and main memory is desired.
+
+    Keyword-only parameters:
+        + :code:`sync_state: gt4py.storage.SyncState`. If `managed="gt4py"` indicates which of the
+          provided buffers, `data` or `device_data`, is up to date at the time of initialization. If
+          the buffers have previously been extracted from a Storage, the :code:`SyncState` object
+          must also be the one extracted from that same original Storage through the
+          :code:`sync_state` attribute. For more details see :ref:`sync_state`.
+
+        For common keyword-only arguments, please see below.
+
+:code:`storage(data: array_like = None, device_data: array_like = None, *, dtype: dtype_like = np.float64, copy=True, **kwargs) -> Storage`
+    Used to allocate a storage with values initialized to those of a given array.
+    If the argument :code:`copy` is set to :code:`False`, the behavior is that of :code:`as_storage`.
```

Comment:
This is to do the same as numpy does. basically, instead of doing `np.asarray(...)` you could do `np.array(..., copy=False)`. 

In the current implementation of this GDP, like numpy, asarray/as_storage is actually just a forwarding to array/storage

https://github.com/numpy/numpy/blob/94721320b1e13fd60046dc8bd0d343c54c2dd2e9/numpy/core/_asarray.py#L85